Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -500,56 +393,163 @@ PID  Name
+416  BTServer
+422  BlueTool
+791  CalendarWidget
+451  CloudKeychainPro
+239  CommCenter
+764  ContactsCoreSpot
 (...)
-21324 ms  -[NSURLRequest initWithURL:0x106388b00 ]
-21324 ms     | -[NSURLRequest initWithURL:0x106388b00 cachePolicy:0x0 timeoutInterval:0x106388b80
 ```
 
+We will demonstrate a few more uses for Frida throughout the chapter.
+
+###### Cycript
+
+Cydia Substrate (formerly called MobileSubstrate) is the standard framework for developing runtime patches ("Cydia Substrate extensions") on iOS. It comes with Cynject, a tool that provides code injection support for C. Cycript is a scripting language developed by Jay Freeman (aka saurik). It injects a JavaScriptCore VM into the running process. Via the Cycript interactive console, users can then manipulate the process with a hybrid Objective-C++ and JavaScript syntax. Accessing and instantiating Objective-C classes inside a running process is also possible. Examples of Cycript usage are included in the iOS chapter.
+
+First download, unpack, and install the SDK.
+
+```shell
+
+#on iphone
+$ wget https://cydia.saurik.com/api/latest/3 -O cycript.zip && unzip cycript.zip
+$ sudo cp -a Cycript.lib/*.dylib /usr/lib
+$ sudo cp -a Cycript.lib/cycript-apl /usr/bin/cycript
+
+```
+
+To spawn the interactive Cycript shell, run "./cyript" or "cycript" if Cycript is on your path.
+
+```shell
+$ cycript
+cy#
+
+```
+
+To inject into a running process, we first need to find the process ID (PID). Running "cycript -p" with the PID injects Cycript into the process. To illustrate, we will inject into SpringBoard.
+
+```shell
+
+$ ps -ef | grep SpringBoard
+501 78 1 0 0:00.00 ?? 0:10.57 /System/Library/CoreServices/SpringBoard.app/SpringBoard
+$ ./cycript -p 78
+cy#
+
+```
+
+We have injected Cycript into SpringBoard. Let's try to trigger an alert message on SpringBoard with Cycript.
+
+```shell
+
+cy# alertView = [[UIAlertView alloc] initWithTitle:@"OWASP MSTG" message:@"Mobile Security Testing Guide"  delegate:nil cancelButtonitle:@"OK" otherButtonTitles:nil]
+#"<UIAlertView: 0x1645c550; frame = (0 0; 0 0); layer = <CALayer: 0x164df160>>"
+cy# [alertView show]
+cy# [alertView release]
+
+```
+
+<img src="Images/Chapters/0x06c/cycript_sample.png" alt="Cycript Alert Sample" width="250">
+
+Find the document directory with Cycript:
+
+```shell
+
+cy# [[NSFileManager defaultManager] URLsForDirectory:NSDocumentDirectory inDomains:NSUserDomainMask][0]
+#"file:///var/mobile/Containers/Data/Application/A8AE15EE-DC8B-4F1C-91A5-1FED35212DF/Documents/"
+
+```
+
+Use the following command to get the application's delegate class:
+
+```shell
+cy# [UIApplication sharedApplication].delegate
+```
+
+The command `[[UIApp keyWindow] recursiveDescription].toString()` returns the view hierarchy of keyWindow. The description of every subview and sub-subview of keyWindow is shown. The indentation space reflects the relationships between views. For example, UILabel, UITextField, and UIButton are subviews of UIView.
```

Comment:
keyWindow with marking? `keyWindow`

Reply:
Same holds for the other classnames

Reply:
I guess this text is already very old haha
I fixed all of them :)